subprojects/core-api/src/main/java/org/gradle/api/provider/CollectionPropertyConfigurer.java

```diff
@@ -70,4 +69,83 @@ public interface CollectionPropertyConfigurer<T> extends ConfigurableValue.Confi
      * @param provider Provider of elements
      */
     void addAll(Provider<? extends Iterable<? extends T>> provider);
+
+    /**
+     * Adds an element to the property value.
+     *
+     * <p>
+     * Contrary to {@link #add(Object)}, this method operates on the
+     * actual value of the property, be it defined via a convention or explicit value.
+     * </p>
+     *
+     * @param element The element
+     * @since 8.7
+     */
+    @Incubating
+    void append(T element);
```

Comment:
❓ Will we deprecate `add*` methods?

Is there any usecase to have both `append` and `add`?

Reply:
Possibly/probably. This PR is focused on getting feedback on the alternative verbs.

Reply:
@asodja Adam brought up the fact that since the alternative API is still incubating, we can't yet deprecate the existing one.

Reply:
Makes sense